File: 2.Valid Anagram/ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
      
        if(s.length() != t.length()){
            return 0;
        }else{
            sort(s.begin(), s.end());
            sort(t.begin(), t.end());  
            if( s != t ){
                return 0;
            }
        }
        return 1;    
    }
};

File: 2.Valid Anagram/ValidAnagram.java
public class ValidAnagram {
    public static void main(String[] args) {
     String s="anagram";
     String t="nagaram";
        System.out.println(isAnagram(s,t));
    }
    public static boolean isAnagram(String s, String t) {
        int l=s.length();
        if(t.length()!=l)
            return false;
        int[] alpha=new int[26];
        for (int i = 0; i <l ; i++) {
            alpha[s.charAt(i) - 'a']++;
            alpha[t.charAt(i)-'a']--;
        }
        for (int i = 0; i < 26; i++) {
            if(alpha[i]!=0)
                return false;
        }
     return true;
    }
}
